substrate-processor/src/interfaces/hooks.ts

```diff
@@ -3,36 +3,76 @@ import {EvmLogHandler, EvmTopicSet} from "./evm"
 import {BlockHandler, EventHandler, ExtrinsicHandler} from "./handlerContext"
 import {QualifiedName} from "./substrate"
 
-
+/**
+ * Defines a {@link BlockHandler} function and the {@link Range} of blocks that should restrain its execution
+ * 
+ * @property handler: {@link BlockHandler}
+ * @property range: {@link Range} (optional)
+ */
 export interface BlockHook {
     handler: BlockHandler
     range?: Range
 }
 
-
+/**
+ * Defines a {@link EventHandler} function, the {@link Range} of blocks, and the Event name that should restrain its execution
+ * 
+ * @property handler: {@link EventHandler}
+ * @property event: {@link QualifiedName} the name of the event that should trigger the handler
+ * @property range: {@link Range} (optional)
+ */
 export interface EventHook {
     handler: EventHandler
     event: QualifiedName
     range?: Range
 }
 
-
+/**
+ * Defines a {@link ExtrinsicHandler} function and the {@link Range} of blocks, Event and Extrinsic names that should restrain its execution
+ * 
+ * @property handler: {@link ExtrinsicHandler}
+ * @property event: {@link QualifiedName} the name of the event that should trigger the handler
+ * @property extrinsic: {@link QualifiedName} the name of the extrinsic that should trigger the handler
+ * @property range: {@link Range} (optional)
+ */
 export interface ExtrinsicHook {
     handler: ExtrinsicHandler
     event: QualifiedName
     extrinsic: QualifiedName
     range?: Range
 }
 
-
+/**
+ * Defines a {@link EvmLogHandler} function and the contract address that should restrain its execution.
```

Comment:
It should be noted when exactly EvmLogHandler is triggered. The following conditions must be satisfied:
- the evm log event was emitted by the given contract address
- the topic of the evm log event matches  `EvmTopicSet` as defined by the [evm event topic convention](https://docs.ethers.io/v5/concepts/events/#events--filters) (if present)
- the block height is in range (if present)